packages/list.lua

```diff
@@ -0,0 +1,107 @@
+SILE.scratch.listStack = {}
+
+SILE.settings.declare({
+  name = "list.bullet.numbered",
+  type = "string",
+  default = ".",
+  help = "Bullet used after a list item number"
+})
+
+-- TODO Not ready to handle RTL
+SILE.settings.declare({
+  name = "list.bullet.inset",
+  type = "Length",
+  default = SILE.length.parse("1.3em"),
+  help = "The amount of space by which each level of the list is shifted"
+})
+
+local defaultBullets = {"•", "◦"}
+for i = 1, #defaultBullets do
+  SILE.settings.declare({
+    name = "list.bullet."..i,
+    type = "string",
+    default = defaultBullets[i],
+    help = "Bullet used for unordered list items at level "..i
+  })
+end
+
+SILE.registerCommand("list", function (options, content)
+  local level = #SILE.scratch.listStack + 1
+  local topLevel = level == 1
+  if topLevel then SILE.call("smallskip") else SILE.typesetter:leaveHmode() end
+
+  local numbered = SU.boolean(options.numbered, SU.boolean(options.ordered, options.numbering ~= nil))
```

Comment:
The mashup of variable names "options.numbered", "options.ordered", and "options.numbering" here combined with nested boolean tests is a bit confusing.

Reply:
I didn't know which one is better. I would pick based on precedent (html - unordered/ordered, tex - ?), but allowing both as an alias could be also fine, users wouldn't have to remember it.

In the grand scheme of things, it could be useful to implement a system which warns for unused options and possibly knows and suggests valid options. But right now, if I specify for example `\list[unordered=false]`, it won't work and I will get no warning. So if I don't notice incorrect output, I am out of luck. But it wouldn't be hard to do warning like: 
> ! Option `unordered` is not used. Did you mean: `ordered`, `numbering` or `bullet`?

That is obviously completely out of scope for this PR, but I could contribute it sometime in the future, if you want.

Reply:
Obscure, not understandable.